platform-emf/src/test/java/com/softicar/platform/emf/form/EmfFormTest.java

```diff
@@ -29,15 +29,44 @@ public void testCreationOfNewEntity() {
 		setInputValue(EmfTestSubObject.NOT_NULLABLE_VALUE, 420);
 		enterNameAndDayAndClickSaveAndClose(SOME_NAME, SOME_DAY);
 
-		// verify results
+		// verify entity
 		assertFalse(entity.impermanent());
 		assertNotNull(entity.getId());
 		assertEquals(SOME_NAME, entity.getName());
 		assertEquals(SOME_DAY, entity.getSimpleEntity().getDay());
+
+		// verify call-backs
+		assertSame(entity, Asserts.assertOne(creationCallbacks));
+
+		// verify UI
 		frame.assertIsClosed();
 		frame.assertIsChangedOnRefreshBus(entity);
 	}
 
+	@Test
+	public void testCreationOfNewEntityWithExceptionInAfterCreationCallback() {
+
+		// create entity and show form
+		EmfTestSubObject entity = showForm(new EmfTestSubObject());
+		form.setCallbackAfterCreation(dummy -> {
+			throw new SofticarUserException(IDisplayString.create("EXCEPTION FROM CALLBACK"));
+		});
+
+		// enter input values and save entity
+		setInputValue(EmfTestSubObject.NOT_NULLABLE_VALUE, 420);
+		enterNameAndDayAndClickSaveAndClose(SOME_NAME, SOME_DAY);
+
+		// verify entity
+		assertFalse(entity.impermanent());
+		assertNotNull(entity.getId());
+		assertEquals(SOME_NAME, entity.getName());
+		assertEquals(SOME_DAY, entity.getSimpleEntity().getDay());
+
+		// verify UI
+		findBody().assertContainsText("EXCEPTION FROM CALLBACK");
```

Comment:
It's great to finally have a test for this.

Reply:
it's exceptional again! :)